sdk_contrib/fetch/lib/fetch_p.js

```diff
@@ -127,7 +128,7 @@ function enableCapture(baseFetchFunction, requestClass, downstreamXRayEnabled, s
       const requestClone = request.clone();
       let response;
       try {
-        response = await baseFetchFunction(requestClone);
+        response = await baseFetchFunction(requestClone, fetchOptions);
```

Comment:
My first idea was to pass down the entire args again but this would overwrite the changes we do to the request like adding the trace header as the headers from the options take priority over the headers that are already set for a request.

Reply:
Consider this code:
```
const agent = new Agent({maxSockets: 1234});
const request = new Request('someurl', { dispatcher: agent });
fetch(request);
```
Here, fetch is only called with one parameter. The dispatcher is "baked in" into the request. For undici, it's actually a symbol property. When you call `baseFetchFunction(requestClone, fetchOptions)`, `requestClone` doesn't contain that property, So that per-request-setting wouldn't be passed on.

Reply:
Made an attempt to fix this. Sorry for the complexity, but it should work more reliably for you.
https://github.com/aws/aws-xray-sdk-node/pull/698

Reply:
#698 fixes a different issue that is there so I would open another issue to point to your PR to.